amplitude.py

```diff
@@ -1,20 +1,125 @@
+import boto3
+import hashlib
+import hmac
 import json
+import os
+import requests
 import sys
+import time
+import zlib
 
-def transform(line):
-    try:
-        parsed_line = json.loads(line)
+AMPLITUDE_API_KEY = os.environ["FXA_AMPLITUDE_API_KEY"]
+HMAC_KEY = os.environ["FXA_AMPLITUDE_HMAC_KEY"]
 
-        assert(type(parsed_line) is dict)
+# For crude pre-emptive rate-limit obedience.
+MAX_EVENTS_PER_BATCH = 10
+MAX_BATCHES_PER_SECOND = 100
+MIN_BATCH_INTERVAL = 1.0 / MAX_BATCHES_PER_SECOND
 
-        event = parsed_line["Fields"]
+def handle (message):
+    # http://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html
+    records = json.loads(message)["Records"]
+    for record in records:
+        if record["eventSource"] != "aws:s3":
+            continue
 
+        s3 = boto3.resource("s3", region_name=record["awsRegion"])
+        s3_object = s3.Object(record["s3"]["bucket"]["name"], record["s3"]["object"]["key"])
+
+        events = ""
+        batch = []
+        for chunk in decompress(s3_object):
+            events += chunk
+            partitioned_events = partition_available_events(events)
+            if is_partitioned(partitioned_events):
+                events = partitioned_events[2]
+                batch = send(partitioned_events[0], batch, False)
+
+        send(events, batch)
+
+def decompress (s3_object):
+    decompressor = zlib.decompressobj(0)
```

Comment:
[The docs](https://docs.python.org/3/library/zlib.html#zlib.decompress) say that passing 0 for `wbits` works for zlib versions >= 1.2.3.5. Is that us?